src/libs/elektra/hooks.c

```diff
@@ -0,0 +1,96 @@
+/**
+* @file
+*
+* @brief
+*
+* @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+*/
+
+#include <kdbinternal.h>
+
+void freeHooks(KDB * kdb, Key * errorKey)
+{
+	if(kdb->hooks == NULL)
+	{
+		return;
+	}
+
+	if(kdb->hooks->gopts != NULL)
+	{
+		elektraPluginClose (kdb->hooks->gopts->plugin, errorKey);
+		free(kdb->hooks->gopts);
+		kdb->hooks->gopts = NULL;
+	}
+
+	free(kdb->hooks);
+	kdb->hooks = NULL;
+}
+
+static struct _HookPluginGopts * initHooksGopts (Plugin * plugin)
+{
+	if(!plugin)
+	{
+		return NULL;
+	}
+
+	struct _HookPluginGopts * hook = elektraCalloc (sizeof(struct _HookPluginGopts));
+	hook->plugin = plugin;
+
+	hook->kdbHookGoptsGet = (kdbHookGoptsGetPtr) elektraPluginGetFunction(plugin, "hooks/gopts/get");
```

Comment:
`elektraPluginGetFunction()` has return type `size_t`. Is there a particual reason why it doesn't return `void *`? This would save us from casting every time we use this function.

Reply:
AFAIK according to the C standard casting function pointers to non-function pointers like `void *` is undefined behaviour (presumably because in theory the Hardware could be Harvard architecture instead of Von-Neumann architecture). Compilers like GCC and Clang allow it and even treat `void (*) (void)` as a special type that is compatible with any function pointer. However all that is non-standard. I assume that's why `elektraPluginGetFunction` returns `size_t`.

At the same time, `dlsym` is defined as `void *dlsym(void *handle, const char *symbol)`, so we may as well use `void *` too. But you'd almost definitely still need a cast, otherwise the man page for `dlsym` wouldn't contain this wonderful snippet:

```c
   /* Writing: cosine = (double (*)(double)) dlsym(handle, "cos");
       would seem more natural, but the C99 standard leaves
       casting from "void *" to a function pointer undefined.
       The assignment used below is the POSIX.1-2003 (Technical
       Corrigendum 1) workaround; see the Rationale for the
       POSIX specification of dlsym(). */

   *(void **) (&cosine) = dlsym(handle, "cos");
```

Reply:
Yeah let's just leave it with size_t :D -- or we maybe could create our own macro, so that the semantics of the return value are clearer. Anyway, that's not a topic for this PR.

´#define ELECTRA_FPTR size_t`